### PR TITLE
Probabilistic

### PR DESCRIPTION
**Pull request by ronuchit**
@tomsilver please review

### CODE REVIEW

File: core.py [hunk 1]
```diff
@@ -51,15 +51,25 @@ def _apply_effects(state, lifted_effects, assignments):
         Maps variables to objects.
     """
     new_literals = set(state.literals)
-
+    determinized_lifted_effects = []
+    # Handle probabilistic effects.
     for lifted_effect in lifted_effects:
+        if isinstance(lifted_effect, ProbabilisticEffect):
+            chosen_effect = lifted_effect.sample()
+            if chosen_effect == "NOCHANGE":
```
Comment (tomsilver) [inline core.py:59]: Why do we need this rather than just an empty set of effects?

File: structs.py [hunk 2]
```diff
@@ -366,6 +367,38 @@ def pddl_str(self):
...
 
+
+class ProbabilisticEffect:
+    """Represents a probabilistic effect over a set of possibilities.
+    """
+    def __init__(self, literals, probabilities):
+        self.literals = literals
+        self.probabilities = probabilities
+        assert sum(self.probabilities) <= 1.0
+        self.literals.append("NOCHANGE")
```
Comment (tomsilver) [inline structs.py:378]: Yeah can we just make this an empty set?

File: wrappers.py [hunk 3]
```diff
@@ -1,96 +0,0 @@
-import gym
```
Comment (tomsilver) [inline wrappers.py:1]: Can you also delete `demo_probabilistic.py` thanks

File: parser.py [hunk 4]
```diff
@@ -289,6 +305,22 @@ def __init__(self, domain_fname, expect_action_preds=True, operators_as_actions=
...
+    def determinize(self):
+        """Determinize this operator by assuming max-probability effects.
+        """
+        assert self.is_probabilistic
+        for op in self.operators.values():
+            toremove = set()
+            for i, lit in enumerate(op.effects.literals):
+                if isinstance(lit, ProbabilisticEffect):
+                    chosen_effect = lit.max()
+                    if chosen_effect == "NOCHANGE":
```
Comment (tomsilver) [inline parser.py:317]: see other comments

**Comment by ronuchit** [comment]
Resolved offline

**Pull request merged**